src/quickgui/qgsquickscalebarkit.h

```diff
@@ -0,0 +1,122 @@
+/***************************************************************************
+  qgsquickscalebarkit.h
+  --------------------------------------
+  Date                 : Nov 2017
+  Copyright            : (C) 2017 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSQUICKSCALEBARKIT_H
+#define QGSQUICKSCALEBARKIT_H
+
+#include <QObject>
+#include <QString>
+
+#include "qgis_quick.h"
+
+class QgsQuickMapSettings;
+
+/**
+ * \ingroup quick
+ *
+ * The class QgsQuickScaleBarKit encapsulates the utilies to calculate
+ * scale bar properties
+ *
+ * It requires connection to mapSettings of the active canvas to automatically
+ * update text and width
+ *
+ * From preferred width in pixel, it calculates the width (pixel) of scalebar
+ * distance in meters or kilometers (int) rounded to "nice" number (e.g. 72.4 to 100)
+ * and units text (e.g. km)
+ *
+ * \note QML Type: ScaleBarKit
+ *
+ * \since QGIS 3.2
+ */
+class QUICK_EXPORT QgsQuickScaleBarKit : public QObject
+{
+    Q_OBJECT
+
+    /**
+      * Associated map settings
+      */
+    Q_PROPERTY( QgsQuickMapSettings *mapSettings MEMBER mMapSettings WRITE setMapSettings NOTIFY mapSettingsChanged )
+
+    /**
+      * Preferred width of scalebar in pixels. Defaults to 300
+      */
+    Q_PROPERTY( int preferredWidth MEMBER mPreferredWidth NOTIFY preferredWidthChanged )
+
+    /**
+      * Units of distance (e.g. km or m) Read-only (result)
```

Comment:
check the properties comments in other qgsquick files to have them in the same format